### PR TITLE
Update kryo library from version 3.0.3 to 4.0.0 and kryo serializers …

### PR DESCRIPTION
**Pull request by JamieGHamilton**
…library from version 0.37 to 0.41.

Modified the builder to support turning field generics off/on and exposed the Kryo Pool from the transcoder if you it for something other than session map serialization.  Use StreamFactory to create Input/output objects.

### CODE REVIEW

File: kryo-serializer/src/main/java/de/javakaffee/web/msm/serializer/kryo/KryoTranscoder.java [hunk 1]
```diff
@@ -206,11 +208,15 @@ private Serializer loadCopyCollectionSerializer( final Class<?> clazz ) {
...
         final Kryo kryo = _kryoPool.borrow();
+        Input in = null;
         try {
-            return kryo.readObject(new Input(data), ConcurrentHashMap.class);
+            in = kryo.getStreamFactory().getInput(data);
+            return kryo.readObject(in, ConcurrentHashMap.class);
         } catch ( final RuntimeException e ) {
             throw new TranscoderDeserializationException( e );
         } finally {
+            IOUtils.closeQuietly(in);
```
Comment (magro) [inline kryo-serializer/src/main/java/de/javakaffee/web/msm/serializer/kryo/KryoTranscoder.java]: I'd prefer our own `closeQuietly` and not to pull in commons-io for this.

File: kryo-serializer/pom.xml [hunk 2]
```diff
@@ -21,12 +21,17 @@
...
+			<version>0.41</version>
 		</dependency>
 		<dependency>
 			<groupId>com.esotericsoftware</groupId>
 			<artifactId>kryo</artifactId>
-			<version>3.0.3</version>
+			<version>4.0.0</version>
+		</dependency>
+		<dependency>
+			<groupId>commons-io</groupId>
```
Comment (magro) [inline kryo-serializer/pom.xml]: Should not be added, instead we should define our own `closeQuietly`.

**Comment by JamieGHamilton** [comment]
Ok, commons-io removed!

**Comment by magro** [comment]
Great, thanks for the PR! Now again, how urgent is a release to you? :smile:

**Comment by JamieGHamilton** [comment]
Be nice to get it soon.  How often can you release?

**Comment by magro** [comment]
Releasing doesn't cost me money, only time - therefore it mainly depends on the pressure of my "day job" and my personal/family schedule... :-)

I'm currently building / preparing the release...

**Comment by JamieGHamilton** [comment]
I see... thank you for doing this release!

**Comment by magro** [comment]
You're welcome :-) 2.1.1 is now on its way to maven central...

**Pull request merged**